src/index.ts

```diff
@@ -1,10 +1,11 @@
-import { DataXY } from 'cheminfo-types';
-import { Shape1D } from 'ml-peak-shape-generator';
+import type { DataXY } from 'cheminfo-types';
+import type { Shape1D } from 'ml-peak-shape-generator';
 import { xMinMaxValues } from 'ml-spectra-processing';
 
-import { getSumOfShapes } from './shapes/getSumOfShapes';
-import { getInternalPeaks } from './util/internalPeaks/getInternalPeaks';
-import { selectMethod } from './util/selectMethod';
+import { getSumOfShapes } from './shapes/getSumOfShapes.ts';
+import { getInternalPeaks } from './util/internalPeaks/getInternalPeaks.ts';
+import { selectMethod } from './util/selectMethod.ts';
+import type { DirectOptimizationOptions as InternalDirectOptimizationOptions } from './util/wrappers/directOptimization.js';
```

Comment:
Avoid using the same interface name in different files. Why not use `InternalDirectOptimizationOptions` in `util/wrappers/directOptimization.js`?